### PR TITLE
Add role and permission guard decorators to some existing routes

### PR DESCRIPTION
**Pull request by cherylli**
# Description

Add role and permission guard decorators to all routes except sprints, resources, and part of ideations, as I would like to look at adding additional permissions for these routes.

TODO - for future, not in this PR
https://github.com/chingu-x/chingu-dashboard-be/pull/112#issuecomment-1988547553

## Issue link

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Review by siasktv — APPROVED** [review]
Wow, this look like a massive amount of work. All tests passed!

File: src/forms/forms.controller.ts [hunk 1]
```diff
@@ -28,6 +31,7 @@ export class FormsController {
         type: FormResponse,
         isArray: true,
     })
+    @Roles(AppRoles.Admin)
```
Comment (timDeHof) [inline src/forms/forms.controller.ts:34]: does role guard need to to be added under Roles decorator?
```suggestion
    @Roles(AppRoles.Admin)
    @UseGuard(RoleGuard)
```

File: src/teams/teams.controller.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class TeamsController {
     constructor(private readonly teamsService: TeamsService) {}
 
     @ApiOperation({
-        summary: "Gets all voyage teams.",
+        summary: "[Roles: Admin] Gets all voyage teams.",
```
Comment (timDeHof) [inline src/teams/teams.controller.ts:34]: Do we have checklist for what is required when writing for swagger?

**Review by timDeHof — CHANGES_REQUESTED** [review]
do we have tests for the guards?

File: src/teams/teams.controller.ts (hunk 2 above)
Comment (cherylli) [reply src/teams/teams.controller.ts:34]: Not atm the moment, but I can write one if you think it would be helpful

**Comment by cherylli** [comment]
> do we have tests for the guards?

yes we should have e2e tests on each endpoint for the permission, but I think we might need to go through each of them and add more soon, as i'm thinking to add more permissions

But I don't think we have tests for the role guard

**Comment by cherylli** [comment]
I decided to only add role check for forms (also did some refactoring while i'm there) because
- ideations: will need to redo to match our current format/standard, too much for this PR
- teams: has no tests yet so it would be another PR itself to add the tests

Resources, sprints, users endpoints I have not touched so I don't intervere with the tests Josh, Curt, and Piero are working on - will need to add decorators to them - also because I want to add more permissions for these endpoints

image

**Comment by cherylli** [comment]
back to draft again - I want to add missing swagger docs to the forms endpoint

**Comment by cherylli** [comment]
ready for review again
image
image

**Comment by cherylli** [comment]
new webhook test

File: test/utils.ts
Comment (timDeHof) [inline test/utils.ts:1]: Great idea of moving the utility functions to a separate file!

**Review by timDeHof — COMMENTED** [review]
I like how you moved the utility functions to a separate file. We should properly do something similar to the function in the other test files.

**Review by curtwl — APPROVED** [review]
Successfully gave 403 for non-admin logged in user

**Review by timDeHof — APPROVED** [review]
great job! looks really clean.

**Pull request merged**